react-ticket-booking-app/src/pages/coupons/coupon-create.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState, useRef } from "react";
+import { ConcertService } from "../../services/concert-service";
+import { CouponService } from "../../services/coupon-service";
+import { useNavigate } from "react-router-dom";
+
+export default function CouponCreate() {
+	var navigate = useNavigate()
+	const couponService = new CouponService()
+	var [concerts, setConcerts] = useState([])
+
+	const concertId = useRef(null)
+	const couponName = useRef(null)
+	const discountPercentage = useRef(null)
+	const coupon = {
+		concertId: '',
+		name: '',
+		discountPercentage: 0
+	}
+
+	useEffect(() => {
+		var concertService = new ConcertService();
+		concertService.getConcertList()
+			.then(data => setConcerts(data.data.concerts))
+			.catch((error) => alert(error.response.data))
+	}, [])
+
+	const createCoupon = (event) => {
+		event.preventDefault()
+		coupon.concertId = concertId?.current?.value || ''
+		coupon.name = couponName?.current?.value || ''
+		coupon.discountPercentage = discountPercentage?.current?.value || 0
+
+		couponService.createCoupon(coupon)
+			.then(() => navigate('../coupons/coupon-list' , { replace: true}))
```

Comment:
string url it is bad practise, if you have 150 place with URL.... move too separate file with all URLs